doc: introduce icon policy documentation

jarolrod opened the pull request:
Moved from https://github.com/bitcoin-core/gui/pull/310

Picking up https://github.com/bitcoin-core/gui/pull/178

This introduces documented policies for how icons should be prepared, optimized, styled, contributed, and attributed within Bitcoin Core. Having guidelines for iconography helps with making applications more consistent and efficient.


[document render](https://github.com/bitcoin-core/gui-qml/blob/23ad199b43e9068d0571eba5965752b03842a7f4/src/qml/doc/icon-policy.md)

hebasto commented on src/qml/doc/icon-policy.md:
```diff
@@ -0,0 +1,39 @@
+and text chunks, resulting in a lossless reduction of the file's size.
+
+## Contributing
+Bitcoin Core primarily uses icons from the [Bitcoin Icon set](https://github.com/BitcoinDesign/Bitcoin-Icons),
```
In the current state of this repo no icons are from the Bitcoin Icon set, right? Maybe clarify that _new_ added icons should be chosen from that set?

jarolrod replied:
We will be using those Icons. Although there are none yet, they will be added as needed. We know that we will primarily use these icons. I don't see the need to state, as of right now, that "new" icons will be chosen from the set.

hebasto commented on src/qml/doc/icon-policy.md:
```diff
@@ -0,0 +1,39 @@
+
+#### Optimizing Production Files (PNG)
+Production (PNG) files must be processed by the [optimize-png.py](https://github.com/bitcoin-core/bitcoin-maintainer-tools/blob/master/optimise-pngs.py) script before their inclusion in Bitcoin Core.
+PNG optimization removes various unnecessary color profiles, ancillary (alla),
```
> ... ancillary (alla), ...

I cannot translate it into my native language ((

jarolrod replied:
https://en.wikipedia.org/wiki/Ancillary_data

jarolrod replied:
i've changed the text here from "ancillary (alla)" to "ancillary data"

hebasto commented on src/qml/doc/icon-policy.md:
```diff
@@ -0,0 +1,39 @@
+#### Optimizing Production Files (PNG)
+Production (PNG) files must be processed by the [optimize-png.py](https://github.com/bitcoin-core/bitcoin-maintainer-tools/blob/master/optimise-pngs.py) script before their inclusion in Bitcoin Core.
+PNG optimization removes various unnecessary color profiles, ancillary (alla),
+and text chunks, resulting in a lossless reduction of the file's size.
```
> ... the file's size

Is this correct from the English Grammar POV?

jarolrod replied:
maybe this is clearer?:

```suggestion
and text chunks, resulting in a lossless file size reduction.
```

hebasto reviewed (commented):
Concept ACK.

shaavan commented on src/qml/doc/icon-policy.md:
```diff
@@ -0,0 +1,39 @@
+
+#### SVG Source File
+SVGs are used as source files because they can scale while retaining image quality.
+They are not used in production due to limited application support.
```
The phrase “limited application support.” sounds a little confusing and might be misleading. For example, one might think, which application are we talking about here? Is it “bitcoin core” or some other application that works on PNG, SVG, and other formats of images.

I think a better framing of this sentence would be:
“They are not used in production because they are not widely supported (compared to PNGs).”

In this framing, the reader will clearly understand that we are talking about format support in general.

jarolrod replied:
This line is just a quick explanation as to why we are not using SVG files as our production files. The reason for this is because of limited application support; this could refer to x, y, or z reason. 

In reality, we are/will be using SVG as production files because that is what designers have recommended. Getting into the particularities here, or what the reality of support for SVG files really is across all applications doesn't matter.

@shaavan: Changing from "limited application support" to "not widely supported" doesn't remove the inherent reference to 'outside' projects

I think the line is fine as is

shaavan replied:
> Changing from "limited application support" to "not widely supported" doesn't remove the inherent reference to 'outside' projects

I can see what you are talking about. I guess the phrase "limited application support" is not as confusing as I first perceived it to be :)

shaavan reviewed (commented):
Concept ACK

The icon policy documentation is quite nicely arranged. The details in the document are pretty well explained without being overwhelming. This writing style fulfills the purpose of the documentation as it would be used chiefly by newcomers in the bitcoin core contributing ecosystem.

I have one change to suggest, which I have mentioned [here](https://github.com/bitcoin-core/gui-qml/pull/70#discussion_r740280447). Other than this and other minor changes suggested by other reviewers, I think this documentation is perfect!

jarolrod commented:
Updated from 23ad199 -> 68efd37 ([pr70.02](https://github.com/jarolrod/gui-qml/tree/pr70.02) -> [pr70.03](https://github.com/jarolrod/gui-qml/tree/pr70.03), [diff](https://github.com/jarolrod/gui-qml/compare/pr70.02..pr70.03))

Changes: 
- Correct grammar on line talking about file size reduction: https://github.com/bitcoin-core/gui-qml/pull/70#discussion_r739946054
- Update link to copyright file from pointing to the old gui repo to this (qml-gui) repo

Bosch-0 commented:
ACK from me :)

hebasto commented on src/qml/doc/icon-policy.md:
```diff
@@ -0,0 +1,39 @@
+
+## Attribution
+Icon additions must include appropriate attribution to the author, license information, and any comments relevant to the icon documented under the
+[contrib/debian/copyright](https://github.com/bitcoin-core/gui-qml/blob/main/contrib/debian/copyright) file.
```
```suggestion
[contrib/debian/copyright](../../../contrib/debian/copyright) file.
```

jarolrod replied:
thanks, updated

hebasto reviewed (approved):
ACK 68efd37c37359ee5fb927deab96eb2964875dfd3

shaavan reviewed (approved):
ACK 68efd37c37359ee5fb927deab96eb2964875dfd3

Thanks again for this fantastic work, @jarolrod!

promag reviewed (commented):
ACK 68efd37c37359ee5fb927deab96eb2964875dfd3.

+1 to @hebasto comment thought.

jarolrod commented:
updated from 68efd37 -> 7335bdd ([pr70.03](https://github.com/jarolrod/gui-qml/tree/pr70.03) -> [pr70.04](https://github.com/jarolrod/gui-qml/tree/pr70.04) ,[diff](https://github.com/jarolrod/gui-qml/compare/pr70.03..pr70.04))

Changes: addressed @hebasto [comment](https://github.com/bitcoin-core/gui-qml/pull/70#discussion_r740758790)

hebasto reviewed (approved):
re-ACK 7335bdd22b0c7a0cbc040a4df373c67fb7e776d6
